lib/pages/feed_page.dart

```diff
@@ -0,0 +1,141 @@
+    return List<int>.from(jsonDecode(response.body));
+  } else {
+    throw Exception('Failed to load album');
+  }
+}
+
+Future<Item> getItem(int itemId) async {
+  final response = await http.get('$baseUrl$item/$itemId$ending');
+
+  if (response.statusCode == 200) {
+    return Item.fromJson(jsonDecode(response.body));
+  } else {
+    throw Exception('Failed to load album');
+  }
+}
+
+class FeedPage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        child: FutureBuilder<List<int>>(
+          future: getItemsIds(newStories),
+          builder: (context, snapshot) {
+            switch (snapshot.connectionState) {
+              case ConnectionState.waiting:
+              case ConnectionState.active:
+                return Center(
+                  child: CircularProgressIndicator(),
+                );
+              case ConnectionState.done:
+                return (snapshot.hasError)
+                    ? Text("${snapshot.error}")
+                    : ListView.builder(
+                        itemCount: snapshot.data.length,
+                        itemBuilder: (BuildContext context, int index) =>
+                            ItemWidget(
+                          snapshot.data[index],
+                          key: Key(snapshot.data[index].toString()),
+                        ),
+                      );
+              default:
+                return Text('Ну это уже слишком');
+            }
+          },
+        ),
+      ),
+    );
+  }
+}
+
+class ItemWidget extends StatelessWidget {
+  final int id;
+
+  ItemWidget(this.id, {Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+      future: getItem(id),
+      builder: (context, snapshot) {
+        switch (snapshot.connectionState) {
+          case ConnectionState.waiting:
+          case ConnectionState.active:
+            return LoadingItem();
+          case ConnectionState.done:
+            return (snapshot.hasError)
+                ? ErrorItem(snapshot.error)
+                : LoadedItem(snapshot.data);
+          default:
+            return ErrorItem(snapshot.connectionState);
+        }
+      },
+    );
+  }
+}
+
+class LoadingItem extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return ListTile(
+      leading: CircularProgressIndicator(),
+      title: Text('Loading...'),
+    );
+  }
+}
+
+class ErrorItem extends StatelessWidget {
+  final Object error;
+
+  ErrorItem(this.error);
+
+  @override
+  Widget build(BuildContext context) {
+    return ListTile(
+      title: Text('Ops... Something went wrong :(\n$error'),
+    );
+  }
+}
+
+class LoadedItem extends StatelessWidget {
+  final Item item;
+
+  LoadedItem(this.item);
+
+  @override
+  Widget build(BuildContext context) {
+    return ListTile(
+      title: Text(item.title),
+      onTap: () {
+        Navigator.push(
```

Comment:
Try to use something more complex, for example, named routes and own router class. (this is not a mistake, just for practice)

Reply:
Replaced with auto_route